tools/clang_tidy/lib/src/command.dart

```diff
@@ -86,20 +89,20 @@ class Command {
     r'//\s*FLUTTER_NOLINT(: https://github.com/flutter/flutter/issues/\d+)?',
   );
 
-  LintAction? _lintAction;
-
   /// The type of lint that is appropriate for this command.
-  Future<LintAction> get lintAction async =>
-    _lintAction ??= await getLintAction(filePath);
+  late final Future<LintAction> lintAction = getLintAction(filePath);
 
   /// Determine the lint action for the file at `path`.
   @visibleForTesting
-  static Future<LintAction> getLintAction(String filePath) {
+  static Future<LintAction> getLintAction(String filePath) async {
     if (path.split(filePath).contains('third_party')) {
-      return Future<LintAction>.value(LintAction.skipThirdParty);
+      return LintAction.skipThirdParty;
     }
 
     final io.File file = io.File(filePath);
+    if (!file.existsSync()) {
```

Comment:
Without this check reading the lines crashes
https://github.com/flutter/engine/blob/ab47e45f55284d68a9107d2d89ae163223245260/tools/clang_tidy/lib/src/command.dart#L103